src/python/pants/backend/go/lint/gofmt/rules.py

```diff
@@ -42,33 +40,27 @@ class GofmtRequest(FmtRequest):
     name = GofmtSubsystem.options_scope
 
 
-@dataclass(frozen=True)
-class SetupRequest:
-    request: GofmtRequest
-    check_only: bool
-
-
 @dataclass(frozen=True)
 class Setup:
     process: Process
     original_snapshot: Snapshot
 
 
 @rule(level=LogLevel.DEBUG)
-async def setup_gofmt(setup_request: SetupRequest, goroot: GoRoot) -> Setup:
+async def setup_gofmt(request: GofmtRequest, goroot: GoRoot) -> Setup:
     source_files = await Get(
         SourceFiles,
-        SourceFilesRequest(field_set.sources for field_set in setup_request.request.field_sets),
+        SourceFilesRequest(field_set.sources for field_set in request.field_sets),
     )
     source_files_snapshot = (
         source_files.snapshot
-        if setup_request.request.prior_formatter_result is None
-        else setup_request.request.prior_formatter_result
+        if request.prior_formatter_result is None
+        else request.prior_formatter_result
```

Comment:
Please!! I can't remember why this was necessary. But it's so copy pasta now.

Reply:
TL;DR Because these rules run for `fmt` (in which case the formatter should run on the prior result) and `lint` (in which case there is no "prior formatter result")